src/App.vue

```diff
@@ -1,12 +1,36 @@
 <script setup lang="ts">
 import '../node_modules/ol/ol.css'
+
+import { watch } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useTranslation } from 'i18next-vue'
+
 import BackgroundSelector from '@/components/background-selector/background-selector.vue'
 import LanguageSelector from '@/components/nav-bars/language-selector.vue'
 import LayerPanel from '@/components/layer-panel/layer-panel.vue'
 import MapContainer from '@/components/map/map-container.vue'
+import { themeSelectorService } from '@/components/theme-selector/theme-selector.service'
+
 import { statePersistorLayersService } from '@/services/state-persistor/state-persistor-layers.service'
+import { statePersistorThemeService } from '@/services/state-persistor/state-persistor-theme.service'
+import { useThemeStore } from '@/stores/config.store'
 
 statePersistorLayersService.bootstrapLayers()
+statePersistorThemeService.bootstrap()
+
+const { t } = useTranslation()
+const themeStore = useThemeStore()
+const { theme } = storeToRefs(themeStore)
+
+watch(
```

Comment:
Why does the `watch` need to be moved into `App` ?

Reply:
Because the theme must be loaded from the url right after arriving on the page.
At this moment, the theme selector component is not mounted (not displayed/not available) and cannot trigger its watchers. This is why I put the watcher here.
But maybe if you have a solution to move this watcher in the theme selector and make it work even if the component is not mounted...?